ns-3_adapter/include/ns-3_adapter.h

```diff
@@ -0,0 +1,196 @@
+#pragma once
+
+#include "ns-3_client.h"
+#include <boost/asio.hpp>
+#include <boost/signals2/signal.hpp>
+#include <boost/lambda/lambda.hpp>
+#include <driver_application/driver_application.h>
+#include <memory>
+#include <mutex>
+#include <condition_variable>
+#include <thread>
+#include <queue>
+#include <vector>
+
+#include <cav_msgs/ByteArray.h>
+#include <cav_srvs/SendMessage.h>
+
+#include <dynamic_reconfigure/server.h>
+
+#include <ros/ros.h>
+
+#include <map>
+#include <set>
+
+/**
+ * @class NS3Adapter
+ * @brief Is the class responsible for the NS-3 Adapter driver
+ */
+class NS3Adapter : public cav::DriverApplication
+{
+    private:
+    struct WaveConfigStruct
+    {
+        std::string name, psid, ns3_id, channel, priority;
+
+        WaveConfigStruct(){};
+        WaveConfigStruct(const std::string &name,
+                         const std::string &psid,
+                         const std::string &ns3_id,
+                         const std::string &channel,
+                         const std::string &priority) : name(name),
+                                         psid(psid),
+                                         ns3_id(ns3_id),
+                                         channel(channel),
+                                         priority(priority) {}
+    };
+
+    public:
+
+        /**
+        * @brief constructor
+        * @param argc - command line argument count
+        * @param argv - command line arguments
+        */
+        NS3Adapter(int argc, char** argv);
+
+        ~NS3Adapter() { shutdown(); }
+
+    private:
+
+        std::vector<std::string> api_;
+
+        //ROS
+        ros::Publisher comms_pub_;
```

Comment:
We'll likely need a few other data streams here as well, I believe. We should be subscribing to the vehicle pose, and we'll also need the vehicle ID (from VehicleConfigParams should be fine) to uniquely identify which vehicle this NS-3 message is associated with (and the pose to say where it's broadcast from). I'd recommend adding subscribers for those whereever you think is most appropriate so that the data can get into that publish NS-3 message function call.